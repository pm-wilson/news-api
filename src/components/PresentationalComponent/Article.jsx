import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ title, author, description, url }) => (
  <div href={url} target="_blank">
    <h3>{title}</h3>
    <span>by {author}</span>
    <p>{description}</p>
  </div>
);

Article.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default Article;
