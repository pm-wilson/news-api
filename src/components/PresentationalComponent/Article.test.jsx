import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Article from './Article';

describe('Article component', () => {
  afterEach(() => cleanup());
  it('renders Article', () => {
    const { asFragment } = render(<Article 
      title="Test Title"
      author="Earthworm Jim"
      description="This is a description of the article"
      url="http://techcrunch.com/2020/10/21/paypal-to-let-you-buy-and-sell-cryptocurrencies-in-the-us/"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
