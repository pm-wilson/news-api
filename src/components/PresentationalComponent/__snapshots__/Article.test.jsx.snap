// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders Article 1`] = `
<DocumentFragment>
  <div
    href="http://techcrunch.com/2020/10/21/paypal-to-let-you-buy-and-sell-cryptocurrencies-in-the-us/"
    target="_blank"
  >
    <h3>
      Test Title
    </h3>
    <span>
      by Earthworm Jim
    </span>
    <p>
      This is a description of the article
    </p>
  </div>
</DocumentFragment>
`;
